### PR TITLE
add initial must-gather script and dockerfile

### PR DESCRIPTION
**Pull request by StevenTobin**
Add initial must-gather script. This script will dump:

- All base k8s resources for fleetshard, strimzi, observability, ingress and any managedkafka namespace
- All observability CRs in the managed-application-services-observability namespace
- All managedkafka and kafka crs
- Ingresscontroller crs
- Additional scrape config secret

To Test:

1. Ping me for cluster details
4. OC login to cluster
5. Run `oc adm must-gather --image=quay.io/stobin/mk-must-gather:0.0.9`
6. Check output of must-gather.local.< UID >

### CODE REVIEW

File: gather_managed_kafka [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#1/bin/bash
```
Comment (obrienrobert) [inline gather_managed_kafka:1]: Never seen a `1` here before. Should this be `!`?

Comment (StevenTobin) [reply gather_managed_kafka:1]: Yes, it should!

**Comment by obrienrobert** [comment]
Do we need to update the README to explain how to run the script and exactly what it will capture?

**Comment by obrienrobert** [comment]
It would be nice to have the `gather_managed_kafka` script formatted. I usually `vi` into the file and run `gg=G` to format it. What do you think?

**Comment by obrienrobert** [comment]
I've run the script a couple of times and it works as expected.

**Comment by StevenTobin** [comment]
> It would be nice to have the `gather_managed_kafka` script formatted. I usually `vi` into the file and run `gg=G` to format it. What do you think?

Is this tabs vs spaces? I use vscode and for bash it's just set to use two spaces. I don't think we have any conventions written down anywhere and running that command in vscode it all looks the same.

**Comment by obrienrobert** [comment]
LGTM.

**Pull request merged**